keras_contrib/applications/nasnet.py

```diff
@@ -618,36 +642,77 @@ def _reduction_A(ip, p, filters, id=None):
 
         h = Activation('relu')(ip)
         h = Conv2D(filters, (1, 1), strides=(1, 1), padding='same', name='reduction_conv_1_%s' % id,
-                   use_bias=False, kernel_initializer='he_normal')(h)
+                   use_bias=False, kernel_initializer='he_normal', kernel_regularizer=l2(weight_decay))(h)
         h = BatchNormalization(axis=channel_dim, momentum=_BN_DECAY, epsilon=_BN_EPSILON,
                                name='reduction_bn_1_%s' % id)(h)
 
         with K.name_scope('block_1'):
-            x1_1 = _separable_conv_block(p, filters, (7, 7), strides=(2, 2), id='reduction_left1_%s' % id)
-            x1_2 = _separable_conv_block(h, filters, (5, 5), strides=(2, 2), id='reduction_right1_%s' % id)
+            x1_1 = _separable_conv_block(p, filters, (7, 7), strides=(2, 2), weight_decay=weight_decay,
+                                         id='reduction_left1_%s' % id)
+            x1_2 = _separable_conv_block(h, filters, (5, 5), strides=(2, 2), weight_decay=weight_decay,
+                                         id='reduction_right1_%s' % id)
             x1 = add([x1_1, x1_2], name='reduction_add_1_%s' % id)
 
         with K.name_scope('block_2'):
             x2_1 = MaxPooling2D((3, 3), strides=(2, 2), padding='same', name='reduction_left2_%s' % id)(h)
-            x2_2 = _separable_conv_block(p, filters, (7, 7), strides=(2, 2), id='reduction_right2_%s' % id)
+            x2_2 = _separable_conv_block(p, filters, (7, 7), strides=(2, 2), weight_decay=weight_decay,
+                                         id='reduction_right2_%s' % id)
             x2 = add([x2_1, x2_2], name='reduction_add_2_%s' % id)
 
         with K.name_scope('block_3'):
             x3_1 = AveragePooling2D((3, 3), strides=(2, 2), padding='same', name='reduction_left3_%s' % id)(h)
-            x3_2 = _separable_conv_block(p, filters, (5, 5), strides=(2, 2), id='reduction_right3_%s' % id)
+            x3_2 = _separable_conv_block(p, filters, (5, 5), strides=(2, 2), weight_decay=weight_decay,
+                                         id='reduction_right3_%s' % id)
             x3 = add([x3_1, x3_2], name='reduction_add3_%s' % id)
 
         with K.name_scope('block_4'):
             x4_1 = MaxPooling2D((3, 3), strides=(2, 2), padding='same', name='reduction_left4_%s' % id)(h)
-            x4_2 = _separable_conv_block(x1, filters, (3, 3), id='reduction_right4_%s' % id)
+            x4_2 = _separable_conv_block(x1, filters, (3, 3), weight_decay=weight_decay, id='reduction_right4_%s' % id)
             x4 = add([x4_1, x4_2], name='reduction_add4_%s' % id)
 
         with K.name_scope('block_5'):
             x5 = AveragePooling2D((3, 3), strides=(1, 1), padding='same', name='reduction_left5_%s' % id)(x1)
 
         x = concatenate([x2, x3, x5, x4], axis=channel_dim, name='reduction_concat_%s' % id)
         return x, ip
+
+
+def _add_auxilary_head(x, classes, weight_decay):
+    '''Adds an auxilary head for training the model
```

Comment:
Can you expand the comment to say why this is desirable or reference the page/section of the paper if it is from there?

Reply:
Added below.